controller_manager:
  ros__parameters:
    update_rate: 
      1000
    joint_group_position_controller:
      type: "position_controllers/JointGroupPositionController"
    trajectory_controller:
      type: "joint_trajectory_controller/JointTrajectoryController"
    joint_state_broadcaster:
      type: "joint_state_broadcaster/JointStateBroadcaster"
    joint_group_effort_controller:
      type: "effort_controllers/JointGroupEffortController"

joint_group_effort_controller:
  ros__parameters:
    joints:
        - frontLeft_hip_motor_joint
        - frontLeft_knee_joint
        - frontLeft_ankle_joint
        - frontRight_hip_motor_joint
        - frontRight_knee_joint
        - frontRight_ankle_joint
        - backLeft_hip_motor_joint
        - backLeft_knee_joint
        - backLeft_ankle_joint
        - backRight_hip_motor_joint
        - backRight_knee_joint
        - backRight_ankle_joint


    command_interfaces:
      - effort

    state_interfaces:
      - position
      - velocity
      - effort
    joint_limit_interface: 
      true
    soft_limits: 
      true
    state_publish_rate: 100.0
    action_monitor_rate: 80.0
    gains:
      default: 
        p: 2.0
        d: 1.5
        i: 0.0
        ff_velocity_scale: 1.0    # <-- ponlo aquÃ­ para silenciar el warning
    constraints:
      goal_time: 0.5
      stopped_velocity_tolerance: 0.05

joint_group_position_controller:
  ros__parameters:
    joints:
      - frontLeft_hip_motor_joint
      - frontLeft_knee_joint
      - frontLeft_ankle_joint
      - frontRight_hip_motor_joint
      - frontRight_knee_joint
      - frontRight_ankle_joint
      - backLeft_hip_motor_joint
      - backLeft_knee_joint
      - backLeft_ankle_joint
      - backRight_hip_motor_joint
      - backRight_knee_joint
      - backRight_ankle_joint
      - articulacion1
      - articulacion2
      - articulacion3
      - articulacion4
      - articulacion5
    command_interfaces:
      - position

    state_interfaces:
      - position
      - velocity

    state_publish_rate: 50.0
    action_monitor_rate: 20.0
    pid_gains:
      default: { p: 50, d: 20, i: 0.0 }
      position: # (or) position
        articulacion1: {kp:  50.0, kd: 20.0, ki:  1.0, max_integral_error: 10000.0}
        articulacion2: {kp:  50.0, kd: 20.0, ki:  1.0, max_integral_error: 10000.0}
        articulacion3: {kp:  50.0, kd: 20.0, ki:  1.0, max_integral_error: 10000.0}
        articulacion4: {kp:  50.0, kd: 20.0, ki:  1.0, max_integral_error: 10000.0}
        articulacion5: {kp:  50.0, kd: 20.0, ki:  1.0, max_integral_error: 10000.0}

trajectory_controller:
  ros__parameters:
    joints:
      - frontLeft_hip_motor_joint
      - frontLeft_knee_joint
      - frontLeft_ankle_joint
      - frontRight_hip_motor_joint
      - frontRight_knee_joint
      - frontRight_ankle_joint
      - backLeft_hip_motor_joint
      - backLeft_knee_joint
      - backLeft_ankle_joint
      - backRight_hip_motor_joint
      - backRight_knee_joint
      - backRight_ankle_joint
      - articulacion1
      - articulacion2
      - articulacion3
      - articulacion4
      - articulacion5
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    joint_limit_interface: true
    soft_limits: true
    state_publish_rate: 50.0
    action_monitor_rate: 20.0
    interpolate_from_desired_state: true
    gains:
      default: { p: 0.001, d: 0.5, i: 0.0 }
    constraints:
      stopped_velocity_tolerance: 0.01
      trajectory: 0.05
      goal:       0.10 
     


joint_state_broadcaster:
  ros__parameters:
    state_publish_rate: 50.0